demos-europe/demosplan-core
# F dplan 12665 adjust phases usage

### Ticket
https://demoseurope.youtrack.cloud/issue/DPLAN-12665/Verfahrensschritte-in-Detailansicht-funktionieren-nicht-richtig

This is a refactor / left over of the ticket above.
- We want to treat phases as objects.
- We want to put in one method, the logic that based on the field "publicStatement" of the stn, the external or internal phases are taken
-- If publicStatement is internal, then internalPhases are taken
-- if publicStatement is external, then externalPhases are taken
- We want to move that logic that I just described above into the BE

In order to do the points above, this PR does the following:
- It has a new StatementPhaseService which does 3 main things:
-- It checks if the phase of the current Statement is a valid one
-- It provides a list of available Phases based on the phase and the publicStatement of the Statement
-- It creates the objects to represent the phases 
-- The method getPhaseName from the StatementService was also adjusted to consume the logic on the StatementPhaseService

### How to review/test
- Trigger api call to check the new fields: phaseStatement in OriginalStn and Stn, 
- Trigger api call to check the new field availablePhases in Stn
- For the availablePhases list: This list returns internal or external Phases based on the "phase" of the stn

### Linked PRs (optional)
Existing PR already merged in integration: [ttps://github.com/demos-europe/demosplan-core/pull/3871/files](https://github.com/demos-europe/demosplan-core/pull/3871/files)

Delete the checkbox if it doesn't apply/isn't necessary.
